src/mon/OSDMonitor.cc

```diff
@@ -5796,17 +5796,53 @@ bool OSDMonitor::prepare_command_impl(MMonCommand *m,
   } else if (prefix == "osd create") {
     int i = -1;
 
+    // optional id provided?
+    int64_t id = -1;
+    if (cmd_getval(g_ceph_context, cmdmap, "id", id)) {
+      if (id < 0) {
+	ss << "invalid osd id value '" << id << "'";
+	err = -EINVAL;
+	goto reply;
+      }
+      dout(10) << " osd create got id " << id << dendl;
+    } else {
+      id = -1;
+    }
+
     // optional uuid provided?
     uuid_d uuid;
     string uuidstr;
     if (cmd_getval(g_ceph_context, cmdmap, "uuid", uuidstr)) {
       if (!uuid.parse(uuidstr.c_str())) {
+	ss << "invalid uuid value '" << uuidstr << "'";
 	err = -EINVAL;
 	goto reply;
       }
       dout(10) << " osd create got uuid " << uuid << dendl;
       i = osdmap.identify_osd(uuid);
-      if (i >= 0) {
+      if (i < 0)
+	i = pending_inc.identify_osd(uuid);
+      if (id >= 0) {
+	// Both id and uuid provided. Check they do not conflict with existing.
+	if (i < 0) {
```

Comment:
indentation seems wrong.  this may be github playing tricks, but please check to make sure that's not a single white-space.